### ISSUE TITLE
Malformed Authorization Header

### ISSUE THREAD

**Issue by bl8dd3r** [aw/picolisp-awscurl#5]
I'm getting this error in response to listing a bucket.
```xml
<?xml version="1.0" encoding="UTF-8"?>
<Error>
  <Code>AuthorizationHeaderMalformed</Code>
  <Message>The authorization header is malformed; a non-empty Access Key (AKID) must be provided in the credential.</Message>
  <RequestId>115013966C11869A</RequestId>
  <HostId>JHBcKMOcEEbiKLZIS7ocMt9DhOGTiSq+yL1Z6coV0HxZxJT0Y8FdI1EClwWC85cJSuvmlymwd9I=</HostId>
</Error>
```
I have my credentials in ~/.aws/credentials in the default section.  I don't use STS and so I don't have an _AWS_SESSION_TOKEN_.
**$HOME/.aws/credentials**
```
[default]
aws_access_key_id = XXXXXXXX
aws_secret_access_key = XXXXXXXX
```

`awscurl.l` works perfectly if the environment has `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` set.
So the creds in the file are good, any idea why they are not being 'sourced' correctly?

**Comment by aw** [comment]
Ah... it seems like it's a bug in the parsing of the file (spaces).

If you set:

```
[default]
aws_access_key_id=XXXXX
```

it should work.. I'll fix the parser. Thanks!

**Comment by aw** [comment]
@bl8dd3r thanks for the bug report! It should be fixed in the latest commit efb21fd

**Comment by bl8dd3r** [comment]
Ahh, the little things! (Issue #2 Add unit tests)
Thanks, This is Awesome!! (<- notice the capital A)
